devtools/dump_devinfo.py

```diff
@@ -191,8 +225,30 @@ async def cli(host, target, basedir, autosave, debug, username, password, batch_
 
     credentials = Credentials(username=username, password=password)
     if host is not None:
-        click.echo("Host given, performing discovery on %s." % host)
-        device = await Discover.discover_single(host, credentials=credentials)
+        if discovery_info:
+            click.echo("Host and discovery info given, trying connect on %s." % host)
+            from kasa import ConnectionType, DeviceConfig
+
+            di = json.loads(discovery_info)
+            dr = DiscoveryResult(**di)
+            connection_type = ConnectionType.from_values(
+                dr.device_type,
+                dr.mgt_encrypt_schm.encrypt_type,
+                dr.mgt_encrypt_schm.lv,
+            )
+            dc = DeviceConfig(
```

Comment:
We should probably add a simple classmethod `create_from_discovery` or similar to simplify this construction, but it's not something for this PR.